### PR TITLE
feat(meta): add source info and stream source split info in get_clust…

### PR DESCRIPTION
**Pull request by KeXiangWang**
…er_info

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Change: Add source info and stream source split info in get_cluster_info interfaces
Background: Orchestrator need source info and source split info to connect to kafka and fetch kakfa status, which is used to estimate tartget rate.
Limitation: 
1. For now, only kafka source is supported and we consider its enough temporarily.
2.  I wasn't sure whether source info and stream source split info should be tackled in ctl, so I did not add anything in ctl. Any suggestions are welcomed.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/meta/src/rpc/service/scale_service.rs [hunk 1]
```diff
@@ -92,9 +106,53 @@ where
...
+            .into_iter()
+            .map(|(id, split_impl_vec)| {
+                let multiple_splits = MultipleSplits {
+                    // Currently orchestrator only support Kafka Source
+                    items: split_impl_vec
+                        .into_iter()
+                        .filter(|split_impl| split_impl.as_kafka().is_some())
+                        .map(|split_impl| {
+                            let (topic, partition) =
+                                split_impl.as_kafka().unwrap().get_topic_and_partition();
```
Comment (BugenZhao) [inline src/meta/src/rpc/service/scale_service.rs:122]:
```suggestion
                        .flat_map(|split_impl| {
                            let (topic, partition) = split_impl.as_kafka()?.get_topic_and_partition();
```

File: proto/meta.proto [hunk 2]
```diff
@@ -208,6 +208,22 @@ message GetClusterInfoRequest {}
 message GetClusterInfoResponse {
   repeated common.WorkerNode worker_nodes = 1;
   repeated TableFragments table_fragments = 2;
+  message KafkaSplit {
```
Comment (BugenZhao) [inline proto/meta.proto:211]: Is there anything we can reuse? cc @tabVersion

**Review by BugenZhao — COMMENTED** [review]
Generally LGTM.

File: src/meta/src/rpc/service/scale_service.rs (hunk 1 above)
Comment (KeXiangWang) [reply src/meta/src/rpc/service/scale_service.rs:122]: It has been filtered, so I think we can use unwrap?

Comment (BugenZhao) [reply src/meta/src/rpc/service/scale_service.rs:122]: With `flat_map` we can check and unwrap in one go. :)

File: proto/meta.proto (hunk 2 above)
Comment (shanicky) [reply proto/meta.proto:211]: Usually we use the `ConnectorSplit` in `source.proto` to represent, can we use the code in connector crate here?

File: src/meta/src/rpc/service/scale_service.rs (hunk 1 above)
Comment (KeXiangWang) [reply src/meta/src/rpc/service/scale_service.rs]: No this problem after using ConnectorSplit

File: proto/meta.proto (hunk 2 above)
Comment (KeXiangWang) [reply proto/meta.proto]: Updated with `ConnectorSplit` in `source.proto`

**Review by shanicky — APPROVED** [review]
LGTM

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**